usr/share/rear/rescue/default/500_ssh.sh

```diff
@@ -3,164 +3,97 @@
-                                    fi
-                                    ;;
-                            esac
-                        done
-                    fi
-
-                    # Carry out the user's decision
-                    case "$passphrase" in
-                        "")
-                            Log "Ignoring unprotected private key $original_file"
-                            rm "$target_file"
-                            ;;
-
-                        UnProtected)
-                            Log "Copying unprotected private key $original_file"
-                            ;;
-
-                        *)
-                            Log "Copying private key $original_file, adding passphrase-protection"
-                            if ! ssh-keygen -p -P '' -N "$passphrase" -f "$target_file"; then
-                                LogPrintError "Could not assign the passphrase to $target_file"
-                                Error "Stopping to avoid leaking unprotected SSH private keys onto the rescue medium"
-                            fi
-                    esac
+local target_dir="$ROOTFS_DIR/root/.ssh" original_file target_file
+
+mkdir -p -m u=rwx,go=- "$target_dir"  # COPY_AS_IS will take effect later, so create the target .ssh directory first
+
+for original_file in /root/.ssh/id_* /root/.ssh/known_host[s]; do
+    target_file="$target_dir/$(basename "$original_file")"
+    cp -p "$original_file" "$target_file"
+
+    case "$original_file" in
+        (*/id_*.pub|*/known_hosts)
+            # Public key files or 'knownhosts' contain no secrets and can be copied as-is
+            Log "Copying $original_file"
+            ;;
+
+        (*/id_*)
+            # Unprotected private key files will be passphrase-protected, if configured, or discarded
+
+            # This ssh-keygen invocation checks if a passphrase change succeeds with an empty passphrase for
+            # authentication. If so, the private key file is not currently passphrase-protected.
+            if ssh-keygen -q -p -P '' -N '' -f "$target_file" >/dev/null 2>&1; then
+                # Handle the unprotected private key file
+
+                if [ -n "$SSH_PRIVATE_KEYS_RECOVER_PASSPHRASE" ]; then
+                    Log "Copying private key $original_file, adding passphrase-protection"
+                    ssh-keygen -p -P '' -N "$SSH_PRIVATE_KEYS_RECOVER_PASSPHRASE" -f "$target_file" || Error "Could not set SSH passphrase for $target_file"
                 else
-                    Log "Copying protected private key $original_file retaining its original passphrase"
-                    LogUserOutput "NOTE: private key $original_file was already passphrase-protected and was copied retaining its original passphrase"
+                    LogPrint "TIP: To have the unprotected SSH private key $original_file copied to the rescue system, see SSH_PRIVATE_KEYS_RECOVER_PASSPHRASE in default.conf"
+                    rm "$target_file"
```

Comment:
So your simplification is to remove the unprotected keys? Are you sure that we can remove this feature (copy unprotected keys to the rescue system) just like this? Maybe users rely on that and use other measures to secure the access (like IP-based restrictions, SSH force commands or even enabling the access only for the time of the recovery).

IMHO this is too harsh. Please add the suggested variable that permits unprotected SSH keys.